python/pyarrow/parquet/core.py

```diff
@@ -3392,6 +3392,8 @@ def write_metadata(schema, where, metadata_collector=None, **kwargs):
     where : string or pyarrow.NativeFile
     metadata_collector : list
         where to collect metadata information.
+    filesystem : FileSystem, default None
+        If passed, will be used in subsequent calls accepting a filesystem.
```

Comment:
That it is being passed to subsequent calls is a bit of an implementation detail, I would say. ParquetWriter has this:

```
filesystem : FileSystem, default None
    If nothing passed, will be inferred from `where` if path-like, else
    `where` is already a file-like object so no filesystem is needed.
```

so maybe let's copy that?